addons/controllers/csi/vspherecsiconfig_utils.go

```diff
@@ -161,8 +168,38 @@ func (r *VSphereCSIConfigReconciler) mapVSphereCSIConfigToDataValuesNonParavirtu
 		dvs.VSphereCSI.PublicNetwork = cpMachineTemplate.Spec.Template.Spec.Network.Devices[0].NetworkName
 	}
 
-	// derive vSphere username and password from the <cluster name> secret
-	clusterSecret, err := cutil.GetSecret(ctx, r.Client, cluster.Namespace, cluster.Name)
+	// derive vSphere username and password from the <cluster name> secret.
+	// In case of TMC based deployment, we do not create <cluster name> secret.
+	// Hence if <cluster name> secret is not found, check for VSphereClusterIdentity resource
+	// to fetch vSphere username and password
+	secretName := cluster.Name
+	secretNamespace := cluster.Namespace
+	for _, variable := range cluster.Spec.Topology.Variables {
```

Comment:
when management cluster is upgraded from legacy (none-cc) mc, will it work as expected?